web/pdf_page_view.js

```diff
@@ -538,22 +538,24 @@ var PDFPageView = (function PDFPageViewClosure() {
       var viewport = pdfPage.getViewport(1);
       // Use the same hack we use for high dpi displays for printing to get
       // better output until bug 811002 is fixed in FF.
-      var PRINT_OUTPUT_SCALE = 2;
+      var DPI = 200; // increase to improve quality
+      var PRINT_OUTPUT_SCALE = DPI/72; 
```

Comment:
There is definitely a trade-off here between memory usage and printing quality, so we need to verify that the memory usage does not become too high. I do think, however, that we should produce good quality prints as that is the whole point of the printing feature (and there are lots of open issues/complaints about the printing quality). I personally think 200 DPI should be fine for a good quality print.

If it turns out that memory usage becomes too high, the alternative solution is to lower the DPI value to 144 such that `DPI / 72` results in 2 again to maintain the same behavior as right now.

/cc @yurydelendik, @brendandahl, @Rob--W for their opinions on this